front/components/assistant_builder/tags/DataSourceTagsFilterDropdown.tsx

```diff
@@ -0,0 +1,261 @@
+          </div>
+        </div>
+      </PopoverContent>
+    </PopoverRoot>
+  );
+}
+
+/**
+ * Handle the search input state logic.
+ */
+function useTagSearch({
+  owner,
+  dustAPIDataSourceIds,
+  selectedTagsIn,
+  selectedTagsNot,
+}: {
+  owner: WorkspaceType;
+  dustAPIDataSourceIds: string[];
+  selectedTagsIn: DataSourceTag[];
+  selectedTagsNot: DataSourceTag[];
+}) {
+  const [searchInputValueIn, setSearchInputValueIn] = useState<string>("");
+  const [searchInputValueNot, setSearchInputValueNot] = useState<string>("");
+  const [availableTagsIn, setAvailableTagsIn] = useState<DataSourceTag[]>([]);
+  const [availableTagsNot, setAvailableTagsNot] = useState<DataSourceTag[]>([]);
+  const searchHandleIn = useRef<NodeJS.Timeout>();
+  const searchHandleNot = useRef<NodeJS.Timeout>();
+  const [isLoadingIn, setIsLoadingIn] = useState(false);
+  const [isLoadingNot, setIsLoadingNot] = useState(false);
+  const { searchTags } = useTagSearchEndpoint({ owner });
+
+  const searchTagsInCoreAPI = async (query: string) => {
+    try {
+      const tags = await searchTags({
+        query,
+        queryType: "prefix",
+        dataSources: dustAPIDataSourceIds,
+      });
+      const formattedTags: DataSourceTag[] = [];
+      for (const tag of tags) {
+        for (const dataSourceId of tag.data_sources) {
+          const isTagUsed =
+            selectedTagsIn.some(
+              (t) => t.tag === tag.tag && t.dustAPIDataSourceId === dataSourceId
+            ) ||
+            selectedTagsNot.some(
+              (t) => t.tag === tag.tag && t.dustAPIDataSourceId === dataSourceId
+            );
+
+          if (!isTagUsed) {
+            formattedTags.push({
+              tag: tag.tag,
+              dustAPIDataSourceId: dataSourceId,
+              connectorProvider: null,
+            });
+          }
+        }
+      }
+      return formattedTags;
+    } catch (error) {
+      console.error("Failed to search tags:", error);
+      return [];
+    }
+  };
+
+  const handleSearchInputChange = (value: string, mode: "in" | "not") => {
+    const setSearchValue =
+      mode === "in" ? setSearchInputValueIn : setSearchInputValueNot;
+    const setAvailableTags =
+      mode === "in" ? setAvailableTagsIn : setAvailableTagsNot;
+    const setIsLoading = mode === "in" ? setIsLoadingIn : setIsLoadingNot;
+    const searchHandle = mode === "in" ? searchHandleIn : searchHandleNot;
+
+    setSearchValue(value);
+
+    if (!value.trim()) {
+      setAvailableTags([]);
+      setIsLoading(false);
+      return;
+    }
+
+    setIsLoading(true);
+    debounce(
+      searchHandle,
+      async () => {
+        try {
+          const formattedTags = await searchTagsInCoreAPI(value);
+          setAvailableTags(formattedTags);
+        } catch (error) {
+          console.error("Failed to search tags:", error);
+          setAvailableTags([]);
+        } finally {
+          setIsLoading(false);
+        }
+      },
+      500
+    );
```

Comment:
shouldn't we skip the search if value is an empty string ?

Reply:
I'm doing it just before this line with 

```
if (!value.trim()) {
      setAvailableTags([]);
      setIsLoading(false);
      return;
    }
```